src/renderer/vt/paint.cpp

```diff
@@ -563,7 +563,7 @@ using namespace Microsoft::Console::Types;
         {
             _deferredCursorPos = { _lastText.X + sNumSpaces, _lastText.Y };
         }
-        else if (numSpaces > 0)
+        else if (numSpaces > 0 && removeSpaces) // if we deleted the spaces... re-add them

```

Comment:
Leave the "why" we're readding them.

Reply:
so part of my confusion is _I actually don't know why we would re-add them_ if we chose to delete them. we should just not choose to delete them.

Reply:
@zadjii-msft can you remember why we'd re-add them if we'd deleted them?

Reply:
Eh if we can't totally suss it out, just leave the MSFT to the code health item here.

Reply:
okay, follow-up filing.

Reply:
I can't remember tbh. This screams organic growth to me - I think we first just decided to do ECH and removing spaces, when we didn't have wrapping (and spaces didn't matter). Then we did some deferring the cursor movement & not erasing optimizations in 20H1, which added cases where we wouldn't ECH. Now we're adding cases where we still need the spaces. I suppose if we'd written it from scratch now, accounting for wrapping, we'd do it better. 

I'd TODO it tbh.

Reply:
https://github.com/microsoft/terminal/issues/5430, linking in the comments

Reply:
I bet we were trying to optimize out the cursor positioning that we'd do after a new line is printed. If we did optimize that scenario, and used ECH to erase some spaces, we also used CUF to move the cursor. However, if we couldn't do that, we'd still want to leave the cursor in the right place...

yea we probably should have just not removed them in the first place.